### PR TITLE
first drafts for upgrading Rails and Ruby in existing projects

### PR DESCRIPTION
**Pull request by rymcmahon**
These are the processes I used for nycl_galaxy. I relied on the Rails docs for upgrading Rails but there wasn't a definitive source of truth for upgrading Ruby, so I just went for it, read the errors, and tried to fix them.

### CODE REVIEW

File: gists/upgrading-rails.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+- It's recommended by the Rails team to update one minor version at a time to make good use of the deprecation warnings. Rails version numbers are in the form `Major.Minor.Patch`. Major and Minor versions are allowed to make changes to the public API, so this may cause errors in your application. Patch versions only include bug fixes, and don't change any public API.
+
+## Process
+
+1. Identify the next minor version of Rails that's above the current version.
+2. Change the version of Rails in the `Gemfile`
+3. If you have gems pinned to specific versions in the `Gemfile`, determine if you still want this behavior. Some of our pinned gems (such as Bourbon and Neat) are not compatible with the latest version of Rails. If you can remove the pins, it's recommended to do so now.
+4. Run `bundle update`. This will update all of the gems in Gemfile to their latest versions unless the version is pinned.
+5. Run `rails app:update`. This will generate new files and change some existing configuration files.
```
Comment (kweingart08) [inline gists/upgrading-rails.md:15]: I don't think I've ever run this. What is this actually updating. Just config files? Like initializers?

**Review by kweingart08 — COMMENTED** [review]
I'm not sure where this goes and if it's relevant to these notes, but I've had to run `gem install rails` after installing a new version of Ruby. Do we need any notes about that?

File: gists/upgrading-rails.md (hunk 1 above)
Comment (rymcmahon) [reply gists/upgrading-rails.md:15]: Yes, it makes changes to the config files and "adds" features that are in the new Rails version. It's from the Upgrading Rails section in the Rails Guides: [https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task](url). I can add that link and explain it further in the documentation if you think that helps?

Comment (kweingart08) [reply gists/upgrading-rails.md:15]: Ya, let's link it. Doesn't hurt to have a link out for further documentation.

**Review by kweingart08 — APPROVED** [review]

**Pull request merged**